### PR TITLE
feat: Redesign quest book page to look similarly to discovery page

### PR DESCRIPTION
**Pull request by kristofbolyai**
Merge https://github.com/Wynntils/Artemis/pull/1778 first.

image

### CODE REVIEW

**Comment by magicus** [comment]
The screenshot indicates that the quest/miniquest selection button in the top is still there. That looks a bit odd, should it not be removed?

File: common/src/main/java/com/wynntils/screens/questbook/WynntilsQuestBookScreen.java [hunk 1]
```diff
@@ -374,7 +359,9 @@ protected void reloadElementsList(String searchText) {
     }
 
     private List<QuestInfo> getSortedQuests() {
-        return miniQuestMode ? Models.Quest.getMiniQuests(contentSortOrder) : Models.Quest.getQuests(contentSortOrder);
+        List<QuestInfo> quests = showQuests ? Models.Quest.getQuests(contentSortOrder) : List.of();
```
Comment (magicus) [inline common/src/main/java/com/wynntils/screens/questbook/WynntilsQuestBookScreen.java]: This will always sort quests before mini-quests. Is that really what you want? If you want to sort by distance, and has both quest and mini-quest toggle enabled, then I'd assume you wanted to see the closest content, regardless of type.

File: common/src/main/java/com/wynntils/screens/questbook/WynntilsQuestBookScreen.java [hunk 2]
```diff
@@ -432,4 +406,59 @@ public void setContentSortOrder(ContentSortOrder newSortOrder) {
         setQuests(getSortedQuests());
         this.setCurrentPage(0);
     }
+
+    private void addQuestProgressTooltipLines(List<Component> tooltipLines, boolean miniQuestMode) {
```
Comment (magicus) [inline common/src/main/java/com/wynntils/screens/questbook/WynntilsQuestBookScreen.java:407]: Something very similar to this information is now available as tooltips from Wynncraft. See e.g. the call chain around DiscoveryModel.updateWorldDiscoveriesFromQuery() on how go extract it.

I'm not sure if it is a suitable replacement, but maybe. (Otoh, changing this can be the fodder of another PR)

File: common/src/main/java/com/wynntils/screens/questbook/widgets/QuestInfoButton.java [hunk 3]
```diff
@@ -6,17 +6,13 @@
 
 import com.mojang.blaze3d.vertex.PoseStack;
 import com.wynntils.screens.base.widgets.WynntilsButton;
-import com.wynntils.screens.questbook.WynntilsQuestBookScreen;
 import com.wynntils.utils.render.RenderUtils;
 import com.wynntils.utils.render.Texture;
 import net.minecraft.network.chat.Component;
 
 public class QuestInfoButton extends WynntilsButton {
```
Comment (magicus) [inline common/src/main/java/com/wynntils/screens/questbook/widgets/QuestInfoButton.java:13]: I think you should remove this entire class. With the filtering, it serves no purpose, and is just confusing.

**Comment by kristofbolyai** [comment]
> The screenshot indicates that the quest/miniquest selection button in the top is still there. That looks a bit odd, should it not be removed?

It shows the progress tooltip you mentioned later.

File: common/src/main/java/com/wynntils/screens/questbook/WynntilsQuestBookScreen.java (hunk 2 above)
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/screens/questbook/WynntilsQuestBookScreen.java:407]: If you don't mind, let's keep it like this now. (I also believe we could add some custom progress info with this code..)

File: common/src/main/java/com/wynntils/screens/questbook/widgets/QuestInfoButton.java (hunk 3 above)
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/screens/questbook/widgets/QuestInfoButton.java:13]: It still shows the progress tooltip, and we can't really render it any other way atm.

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/screens/questbook/widgets/QuestInfoButton.java:13]: image

File: common/src/main/java/com/wynntils/screens/questbook/WynntilsQuestBookScreen.java (hunk 1 above)
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/screens/questbook/WynntilsQuestBookScreen.java]: I wondered why it looked weird.. Fixed.

File: common/src/main/java/com/wynntils/screens/questbook/WynntilsQuestBookScreen.java (hunk 2 above)
Comment (magicus) [reply common/src/main/java/com/wynntils/screens/questbook/WynntilsQuestBookScreen.java:407]: Yes, it was more a case of "oh, this reminds me of...".

File: common/src/main/java/com/wynntils/screens/questbook/widgets/QuestInfoButton.java (hunk 3 above)
Comment (magicus) [reply common/src/main/java/com/wynntils/screens/questbook/widgets/QuestInfoButton.java:13]: Ah, I see.

**Review by magicus — APPROVED** [review]
